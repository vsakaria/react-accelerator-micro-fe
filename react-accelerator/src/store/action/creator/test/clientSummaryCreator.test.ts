import {
  clientSummaryRequest,
  clientSummarySuccess,
  clientSummaryFailure
} from "../clientSummaryCreator";

describe("ClientSummary Action Creator", () => {
  it("should creator a request action", () => {
    const clientSummaryRequestAction = clientSummaryRequest();
    expect(clientSummaryRequestAction).toMatchSnapshot();
  });

  it("should creator a success action", () => {
    const payload = {
      data: "some_data"
    };
    const clientSummaryRequestAction = clientSummarySuccess(payload);
    expect(clientSummaryRequestAction).toMatchSnapshot();
  });

  it("should creator a failure action", () => {
    const error = new Error("SOME ERROR MESSAGE");
    const clientSummaryRequestAction = clientSummaryFailure(error);
    expect(clientSummaryRequestAction).toMatchSnapshot();
  });
});
