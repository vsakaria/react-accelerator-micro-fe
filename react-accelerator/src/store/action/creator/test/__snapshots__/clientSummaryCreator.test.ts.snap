// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClientSummary Action Creator should creator a failure action 1`] = `
Object {
  "data": Object {
    "error": true,
    "loading": false,
    "message": "SOME ERROR MESSAGE",
  },
  "type": "CLIENT_SUMMARY_FAILURE",
}
`;

exports[`ClientSummary Action Creator should creator a request action 1`] = `
Object {
  "data": Object {
    "loading": true,
  },
  "type": "CLIENT_SUMMARY_REQUEST",
}
`;

exports[`ClientSummary Action Creator should creator a success action 1`] = `
Object {
  "data": Object {
    "data": "some_data",
    "loading": false,
  },
  "type": "CLIENT_SUMMARY_SUCCESS",
}
`;
